Makefile

```diff
@@ -27,17 +27,6 @@ style: ## Run the code styler
 		! -path Cabal/src/Distribution/Simple/Build/PathsModule/Z.hs \
 		| xargs -P $(PROCS) -I {} fourmolu -q -i {}
 
-# source generation: Lexer
-
-LEXER_HS:=Cabal-syntax/src/Distribution/Fields/Lexer.hs
-
-lexer : $(LEXER_HS)
-
-$(LEXER_HS) : templates/Lexer.x
-	alex --latin1 --ghc -o $@ $^
```

Comment:
I don't know if `--ghc` is the default in Alex, but also, I don't know how important this optimization is for boot-strapping.
There is a 10 year old feature request in Alex that would allow you to pass this flag in via the `.x` file, maybe time to implement this:
- https://github.com/haskell/alex/issues/25

How does `cabal` call `alex` anyway by default?  What flags are given?  
I found this TODO comment: https://github.com/haskell/cabal/blob/be2fd1adf356caeef65c54febefb2b6da7dff36e/Cabal/src/Distribution/Simple/PreProcess.hs#L120-L128

Ok, now I found that flag `-g` is hardcoded if `alex` is used with `ghc`: https://github.com/haskell/cabal/blob/be2fd1adf356caeef65c54febefb2b6da7dff36e/Cabal/src/Distribution/Simple/PreProcess.hs#L885-L892

So, it seems that we need not worry about the `--ghc` flag.

Reply:
Thanks for fishing this out!